aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -116,9 +118,49 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         raise NotImplementedError()
 
 
+@overload
+def get_secret(
+    name: str,
+    transform: None = None,
+    force_fetch: bool = False,
+    max_age: Optional[int] = None,
+    **sdk_options,
+) -> str: ...
+
+
+@overload
+def get_secret(
+    name: str,
+    transform: Literal["json"],
+    force_fetch: bool = False,
+    max_age: Optional[int] = None,
+    **sdk_options,
+) -> dict: ...
+
+
+@overload
+def get_secret(
+    name: str,
+    transform: Literal["binary"],
+    force_fetch: bool = False,
+    max_age: Optional[int] = None,
+    **sdk_options,
+) -> Union[str, dict, bytes]: ...
+
+
+@overload
+def get_secret(
+    name: str,
+    transform: Literal["auto"],
+    force_fetch: bool = False,
+    max_age: Optional[int] = None,
+    **sdk_options,
+) -> bytes: ...
+
+
 def get_secret(
     name: str,
-    transform: Optional[str] = None,
+    transform: TransformOptions = None,
```

Comment:
Nice catch @TonySherman! My IDE can now complete the possible values for this field, amazing!